app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchTEActivity.java

```diff
@@ -984,6 +988,49 @@ public boolean onMapClick(@NonNull LatLng point) {
         return findFeature(rectF, sourcesAndLayer.component1(), sourcesAndLayer.component2(), 0);
     }
 
+    @Override
+    public boolean onMapLongClick(@NonNull LatLng point) {
+        PointF pointf = teiMapManager.getMap().getProjection().toScreenLocation(point);
+        RectF rectF = new RectF(pointf.x - 10, pointf.y - 10, pointf.x + 10, pointf.y + 10);
+
+        Pair<List<String>, List<String[]>> sourcesAndLayer = teiMapManager.getSourcesAndLayersForSearch();
+        String source = sourcesAndLayer.component1().get(0);
+        boolean featureFound = false;
+        if (source.contains(TEI)) {
+            featureFound = findFeature(rectF, sourcesAndLayer.component1(), sourcesAndLayer.component2(), 0);
+        }
+        if (featureFound) {
+            Feature feature = getFeature(rectF, TEI, "TEI_POINT_LAYER_ID");
+            if (feature != null) {
+                navigateToMap(feature);
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private void navigateToMap(Feature feature) {
+        LatLng point = FeatureExtensionsKt.getPointLatLng(feature);
+        String longitude = String.valueOf(point.getLongitude());
+        String latitude = String.valueOf(point.getLatitude());
+        String location = "geo:0,0?z=11&q=" + latitude + "," + longitude + "";
+
+        Intent intent = new Intent(Intent.ACTION_VIEW);
+        intent.setData(Uri.parse(location));
+        startActivity(intent);
+    }
+
+    private Feature getFeature(RectF rectF, String source, String layer) {
+        List<Feature> features = teiMapManager.getMap().queryRenderedFeatures(rectF, layer);
+        Feature feature = null;
+        if (!features.isEmpty()) {
+            if (source.contains(TEI)) {
+                feature = features.get(0);
+            }
+        }
+        return feature;
+    }
```

Comment:
Most of this could be moved to the map package in a controller for example. Then here you could implement your controller listener, doing this search activity would not know about mapbox and we have this feature in map package, otherwise if we convert the map package to library we will loose this. Also, you could test it too :+1:

Reply:
You are right, I'll go into doing this! 👍

Reply:
you can leave the intent part here :+1: